packages/filecoin-actor-utils/src/utils.ts

```diff
@@ -0,0 +1,45 @@
+import { networkActorCodeMap } from "./data"
+
+/**
+ * Resolves the actor name by providing the actor code. When also providing
+ * the network name, it will limit the search to that specific network.
+ * @param actorCode the actor code which is used to resolve the actor name
+ * @param networkName (optional) the network in which to search for the actor name 
+ * @returns the actor name when found or null when not found
+ */
+export const getActorName = (actorCode: string, networkName?: string): string | null => {
+  
+  // If "networkName" is provided, find the
+  // actor name for that specific network
+  if (networkName) {
+    const actorCodeMap = networkActorCodeMap[networkName]
+    for (const [name, code] of Object.entries(actorCodeMap)) {
+      if (code === actorCode) return name
+    }
+  }
+  
+  // Otherwise, iterate over all the
+  // networks to find the actor name
+  else {
```

Comment:
When would be a time that we don't have the network name?

Reply:
Not sure, it could be a convenience thing. Just give any actor code and it finds the name regardless of the network.
But fair point, we can totally make it mandatory to provide the network! Let me know what you prefer, for me it's fine either way.